### PR TITLE
fix(gsd): prevent ghost milestone directories from becoming active

### PR DESCRIPTION
**Pull request by Jamie-BitFlight**
## TL;DR

**What:** Prevent ghost milestone directories from becoming the active milestone during state derivation and doctor repair.
**Why:** Empty or metadata-only milestone directories (created by doctor repair or manual mkdir) could outrank real in-flight milestones in sequence-based election, redirecting auto-mode to fabricated earlier milestones.
**How:** Add `isSubstantiveMilestone()` filtering to milestone discovery, ghost milestone warnings to doctor, and regression tests covering the full recovery path.

## What

Ghost milestone directories (empty directories under `.gsd/milestones/`) were being included in milestone discovery and could become the active milestone in `deriveState()`. This caused auto-mode to redirect away from the user's actual in-flight work after doctor repair.

This PR:

1. Adds `isSubstantiveMilestone()` to `milestone-ids.ts` — checks for ROADMAP, CONTEXT, CONTEXT-DRAFT, SUMMARY, or populated slices/ before treating a directory as a real milestone
2. Filters `findMilestoneIds()` through that check so ghost directories are excluded from state derivation
3. Adds `orphaned_milestone_directory` warnings to doctor so users can see and clean up ghost directories
4. Re-exports `isSubstantiveMilestone` through `guided-flow.ts` for backward compatibility
5. Adds two focused regression test files:
   - `ghost-milestone-regression.test.ts` — 25 assertions covering discovery, state derivation, and edge cases
   - `doctor-ghost-regression.test.ts` — 21 assertions covering doctor warnings, rebuildState, and full repair flow

## Why

A real user incident: damaged `.gsd/` state combined with doctor repair created empty milestone directories that outranked the real in-flight milestone during `deriveState()` sequence-based election. Auto-mode then resumed on a fabricated earlier milestone instead of the user's actual work.

## How

- `isSubstantiveMilestone()` is a pure filesystem check — no side effects, no state mutation
- `findMilestoneIds()` filters through it before returning, so all downstream consumers (state derivation, doctor, guided flow) automatically exclude ghosts
- Doctor scans raw milestone directories separately and emits `orphaned_milestone_directory` warnings for non-substantive directories, giving users visibility without silently deleting anything
- The function lives in `milestone-ids.ts` (where discovery logic belongs in current main), not in `guided-flow.ts`

## Change type

- [x] `fix` — Bug fix
- [x] `test` — Adding or updating tests

## Scope

- [x] `gsd extension` — GSD workflow

## Breaking changes

- [x] No breaking changes

## Test plan

- [ ] CI passes
- [x] New/updated tests included
- [x] Manual testing — steps described above

Local verification:
```
NODE_OPTIONS='--import=./src/resources/extensions/gsd/tests/resolve-ts-hooks.mjs' \
  npx tsx --test \
  src/resources/extensions/gsd/tests/ghost-milestone-regression.test.ts \
  src/resources/extensions/gsd/tests/doctor-ghost-regression.test.ts
```
Result: 46 passed, 0 failed.

## AI disclosure

- [x] This PR includes AI-assisted code

### CODE REVIEW

File: src/resources/extensions/gsd/tests/ghost-milestone-regression.test.ts [hunk 1]
```diff
@@ -0,0 +1,403 @@
+// Regression test: ghost milestone directories should not become active.
+//
+// Issue: When empty "ghost" milestone directories (M001, M002) exist alongside
+// a real milestone with actual content (M005), deriveState() incorrectly
+// activates the ghost M001 instead of the real M005.
+//
+// This test should FAIL against the current codebase, proving the bug exists.
+// After T02 implements ghost rejection, this test should PASS.
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/tests/ghost-milestone-regression.test.ts]: Fixed — updated header comments to describe the intended invariant instead of referencing a future task. Commit 61357ec0.

File: src/resources/extensions/gsd/milestone-ids.ts [hunk 2]
```diff
@@ -72,6 +73,42 @@ export function nextMilestoneId(milestoneIds: string[], uniqueEnabled?: boolean)
...
 // ─── Discovery ──────────────────────────────────────────────────────────────
 
+/**
+ * Check if a milestone directory contains substantive content.
+ * Ghost milestones (empty directories or metadata-only skeletons) should be
+ * filtered out so they never become the active milestone.
+ */
+export function isSubstantiveMilestone(basePath: string, milestoneId: string): boolean {
+  const milestoneDir = join(basePath, ".gsd", "milestones", milestoneId);
+
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/milestone-ids.ts:122]: Fixed — now uses milestonesDir() instead of hardcoded .gsd path. Commit 61357ec0.

File: src/resources/extensions/gsd/guided-flow.ts [hunk 3]
```diff
@@ -32,21 +32,26 @@ import { showProjectInit, offerMigration } from "./init-wizard.js";
 import { validateDirectory } from "./validate-directory.js";
 import { showConfirm } from "../shared/mod.js";
 import { debugLog } from "./debug-logger.js";
-import { findMilestoneIds, nextMilestoneId } from "./milestone-ids.js";
+import { MILESTONE_ID_RE, findMilestoneIds, isSubstantiveMilestone } from "./milestone-ids.js";
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/guided-flow.ts]: Fixed — added nextMilestoneId to the import from milestone-ids.js. Commit 61357ec0.

File: src/resources/extensions/gsd/tests/dist-redirect.mjs [hunk 4]
```diff
@@ -1,17 +1,32 @@
 const ROOT = new URL("../../../../../", import.meta.url);
+const MAIN_REPO = new URL("file:///home/ubuntulinuxqa2/repos/gsd-2/");
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/tests/dist-redirect.mjs]: Fixed — MAIN_REPO now derives from ROOT with env var override. Commit 61357ec0.

File: src/resources/extensions/gsd/tests/doctor-runtime.test.ts [hunk 5]
```diff
@@ -238,7 +238,10 @@ None
 
       // Verify .gsd entry was added (external state symlink)
       const content = readFileSync(join(dir, ".gitignore"), "utf-8");
-      assertTrue(content.includes(".gsd"), "gitignore now has .gsd entry");
+      assertTrue(content.includes(".gsd/activity/"), "gitignore now has activity pattern");
+      assertTrue(content.includes(".gsd/auto.lock"), "gitignore now has auto.lock pattern");
+      assertTrue(content.includes(".gsd/gsd.db-wal"), "gitignore now has gsd.db-wal pattern");
+      assertTrue(content.includes(".gsd/gsd.db-shm"), "gitignore now has gsd.db-shm pattern");
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/tests/doctor-runtime.test.ts:244]: These test assertions match existing behavior from main's ensureGitignore(). No change needed.

File: src/resources/extensions/gsd/tests/doctor-git.test.ts [hunk 6]
```diff
@@ -218,9 +220,13 @@ async function main(): Promise<void> {
...
-      // Verify file is no longer tracked
-      const tracked = run("git ls-files .gsd/activity/", dir);
-      assertEq(tracked, "", "runtime file untracked after fix");
+      // Verify files are no longer tracked
+      const trackedActivity = run("git ls-files .gsd/activity/", dir);
+      assertEq(trackedActivity, "", "activity runtime file untracked after fix");
+      const trackedWal = run("git ls-files .gsd/gsd.db-wal", dir);
+      assertEq(trackedWal, "", "gsd.db-wal untracked after fix");
+      const trackedShm = run("git ls-files .gsd/gsd.db-shm", dir);
+      assertEq(trackedShm, "", "gsd.db-shm untracked after fix");
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/tests/doctor-git.test.ts:229]: These test assertions match existing RUNTIME_EXCLUSION_PATHS behavior on main. No change needed.

File: src/resources/extensions/gsd/doctor.ts [hunk 7]
```diff
@@ -350,7 +358,505 @@ export async function selectDoctorScope(basePath: string, requestedScope?: strin
...
+        message: `${trackedPaths.length} runtime file(s) are tracked by git: ${trackedPaths.slice(0, 5).join(", ")}${trackedPaths.length > 5 ? "..." : ""}`,
+        fixable: true,
+      });
+
+      if (shouldFix("tracked_runtime_files")) {
+        try {
+          for (const exclusion of RUNTIME_EXCLUSION_PATHS) {
+            nativeRmCached(basePath, [exclusion]);
+          }
+          fixesApplied.push(`untracked ${trackedPaths.length} runtime file(s)`);
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/doctor.ts]: Carried over from main — the exclusion paths already cover these patterns.

File: src/resources/extensions/gsd/doctor.ts [hunk 8]
```diff
@@ -350,7 +358,505 @@ export async function selectDoctorScope(basePath: string, requestedScope?: strin
...
+            unitId: "project",
+            message: `${missing.length} critical GSD runtime pattern(s) missing from .gitignore: ${missing.join(", ")}`,
+            file: ".gitignore",
+            fixable: true,
+          });
+
+          if (shouldFix("gitignore_missing_patterns")) {
+            ensureGitignore(basePath);
+            fixesApplied.push("added missing GSD runtime patterns to .gitignore");
+          }
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/doctor.ts]: Carried over from main — ensureGitignore() already handles these patterns.

**Comment by glittercowboy** [comment]
Merged — ghost milestone detection implemented via isGhostMilestone() in state.ts (5acbce649).
